File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minP=prices[0];
        int maxP=0;
        for (int i=0;i<prices.size();i++){
            minP=min(minP,prices[i]);
            maxP=max(maxP,prices[i]-minP);
        }
        return maxP;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxP,curr=0,prices[0]
        for i in range(0,len(prices)):
            curr=min(curr,prices[i])
            maxP=max(maxP,prices[i]-curr)
        return maxP